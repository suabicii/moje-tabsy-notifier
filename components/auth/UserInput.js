import React from "react";
import {TextInput, View} from "react-native";
import {Text} from "react-native";

function UserInput({name, value, setValue, keyboardType = "default", secureTextEntry = false}) {
    return (
        <View style={{marginHorizontal: 24}}>
            <Text style={{
                color: "#414959",
                fontWeight: "300"
            }}>
                {name}
            </Text>
            <TextInput
                keyboardType={keyboardType}
                secureTextEntry={secureTextEntry}
                style={{
                    borderBottomWidth: 0.5,
                    height: 48,
                    borderBottomColor: '#8e93a1',
                    marginBottom: 30
                }}
                value={value}
                onChangeText={text => setValue(text)}
            />
        </View>
    );
}

export default UserInput;