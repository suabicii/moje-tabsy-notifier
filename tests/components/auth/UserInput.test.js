import React from "react";
import renderer from "react-test-renderer";
import UserInput from "../../../components/auth/UserInput";

it('should correctly render UserInput component', () => {
    const setValue = jest.fn();
    const tree = renderer.create(
        <UserInput name="some_name" value="some_value" setValue={setValue}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});