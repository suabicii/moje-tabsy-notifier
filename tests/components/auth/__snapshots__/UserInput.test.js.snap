// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render UserInput component 1`] = `
<View
  style={
    {
      "marginHorizontal": 24,
    }
  }
>
  <Text
    style={
      {
        "color": "#414959",
        "fontWeight": "300",
      }
    }
  >
    some_name
  </Text>
  <TextInput
    keyboardType="default"
    onChangeText={[Function]}
    secureTextEntry={false}
    style={
      {
        "borderBottomColor": "#8e93a1",
        "borderBottomWidth": 0.5,
        "height": 48,
        "marginBottom": 30,
      }
    }
    value="some_value"
  />
</View>
`;
